src/main/java/com/mraof/minestuck/block/redstone/AreaEffectBlock.java

```diff
@@ -0,0 +1,92 @@
+package com.mraof.minestuck.block.redstone;
+
+import com.mraof.minestuck.client.gui.MSScreenFactories;
+import com.mraof.minestuck.effects.CreativeShockEffect;
+import com.mraof.minestuck.tileentity.redstone.AreaEffectTileEntity;
+import com.mraof.minestuck.util.ParticlesAroundSolidBlock;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.PotionItem;
+import net.minecraft.particles.RedstoneParticleData;
+import net.minecraft.potion.EffectInstance;
+import net.minecraft.potion.PotionUtils;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.ActionResultType;
+import net.minecraft.util.Hand;
+import net.minecraft.util.SoundCategory;
+import net.minecraft.util.SoundEvents;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.BlockRayTraceResult;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraft.world.IBlockReader;
+import net.minecraft.world.World;
+
+import javax.annotation.Nullable;
+import java.util.Random;
+
+public class AreaEffectBlock extends Block
+{
+	public static final String EFFECT_CHANGE_MESSAGE = "effect_change_message";
+	
+	public AreaEffectBlock(Properties properties)
+	{
+		super(properties);
+	}
+	
+	@Override
+	public boolean hasTileEntity(BlockState state)
+	{
+		return true;
+	}
+	
+	@Nullable
+	@Override
+	public TileEntity createTileEntity(BlockState state, IBlockReader world)
+	{
+		return new AreaEffectTileEntity();
+	}
+	
+	@Override
+	public ActionResultType use(BlockState state, World worldIn, BlockPos pos, PlayerEntity player, Hand hand, BlockRayTraceResult hit)
+	{
+		if(player.isCreative() && !CreativeShockEffect.doesCreativeShockLimit(player, 1, 4))
```

Comment:
Creative Shock is intended to prevent access to certain GUIs, at a strong enough amplifier(5) this even applies to creative mode players

Reply:
This does not have a gui but it is still intended to prevent changing the effect

Reply:
Alright, slightly different question: Why should players in creative mode be affected by creative shock at all?

Reply:
I thought it would be useful in situations like needing spawn protection, or in multiplayer settings for preventing people in creative mode without op capabilities from modifying an environment if desired

Reply:
I don't think an effect is well suited to use as a permission system for creative mode. As such, I don't think we should design it to be used in that way.
While I think that it is a questionable choice to have the effect apply to creative mode players, I don't think it's an important choice, so I'm fine with it if you still want to keep it.

Reply:
Where do you want to go from here? Has your mind changed any, or do you want to leave this as it is?

Reply:
Unless a major issue is discovered I would like to keep it as is. I am open to removing it with additional feedback